src/FirebirdSql.Data.FirebirdClient.Tests/FbDatabaseInfoTests.cs

```diff
@@ -61,4 +62,119 @@ public void CompleteDatabaseInfoTest()
+		IDictionary<short, ulong> readSeqCount = dbInfo.GetReadSeqCount();
+		IDictionary<short, ulong> readIdxCount = dbInfo.GetReadIdxCount();
+
+		var fbCommand = new FbCommand("INSERT INTO TEST (INT_FIELD) VALUES (900)", Connection);
+		fbCommand.ExecuteNonQuery();
+
+		insertCount = GetAffectedTables(insertCount, dbInfo.GetInsertCount());
+		updateCount = GetAffectedTables(updateCount, dbInfo.GetUpdateCount());
+		readSeqCount = GetAffectedTables(readSeqCount, dbInfo.GetReadSeqCount());
+		readIdxCount = GetAffectedTables(readIdxCount, dbInfo.GetReadIdxCount());
+
+		Assert.That(insertCount.ContainsKey(tableIdTest), Is.True);
+		Assert.That(insertCount[tableIdTest], Is.EqualTo(1));
+		Assert.That(updateCount.ContainsKey(tableIdTest), Is.False);
+		Assert.That(readSeqCount.ContainsKey(tableIdTest), Is.False);
+		Assert.That(readIdxCount.ContainsKey(tableIdTest), Is.False);
+	}
+
+	[Test]
+	public void PerformanceAnalysis_UPDATE_Test()
+	{
+		IDictionary<string, short> tableNameList = GetTableNameList();
+		short tableIdTest = tableNameList["TEST"];
+
+		var fbCommand = new FbCommand("INSERT INTO TEST (INT_FIELD) VALUES (900)", Connection);
+		fbCommand.ExecuteNonQuery();
+
+		var dbInfo = new FbDatabaseInfo(Connection);
+		IDictionary<short, ulong> insertCount = dbInfo.GetInsertCount();
+		IDictionary<short, ulong> updateCount = dbInfo.GetUpdateCount();
+		IDictionary<short, ulong> readSeqCount = dbInfo.GetReadSeqCount();
+		IDictionary<short, ulong> readIdxCount = dbInfo.GetReadIdxCount();
+
+		fbCommand.CommandText = "UPDATE TEST SET SMALLINT_FIELD = 900 WHERE (INT_FIELD = 900)";
+		fbCommand.ExecuteNonQuery();
+
+		insertCount = GetAffectedTables(insertCount, dbInfo.GetInsertCount());
+		updateCount = GetAffectedTables(updateCount, dbInfo.GetUpdateCount());
+		readSeqCount = GetAffectedTables(readSeqCount, dbInfo.GetReadSeqCount());
+		readIdxCount = GetAffectedTables(readIdxCount, dbInfo.GetReadIdxCount());
+
+		Assert.That(insertCount.ContainsKey(tableIdTest), Is.False);
+		Assert.That(updateCount.ContainsKey(tableIdTest), Is.True);
+		Assert.That(updateCount[tableIdTest], Is.EqualTo(1));
+		Assert.That(readSeqCount.ContainsKey(tableIdTest), Is.False);
+		Assert.That(readIdxCount.ContainsKey(tableIdTest), Is.True);
+		Assert.That(readIdxCount[tableIdTest], Is.EqualTo(1));
+	}
+
+	private IDictionary<short, ulong> GetAffectedTables(IDictionary<short, ulong> aStatisticInfoBefore, IDictionary<short, ulong> aStatisticInfoAfter)
+	{
+		var result = new Dictionary<short, ulong>();
+		foreach (KeyValuePair<short, ulong> keyValuePair in aStatisticInfoAfter)
+		{
+			if (aStatisticInfoBefore.TryGetValue(keyValuePair.Key, out ulong value))
+			{
+				ulong counter = keyValuePair.Value - value;
+				if (counter > 0)
+					result.Add(keyValuePair.Key, counter);
+			}
+			else
+				result.Add(keyValuePair.Key, keyValuePair.Value);
+		}
+		return result;
+	}
+
+	private IDictionary<string, short> GetTableNameList()
```

Comment:
I think it would be better to use `FbSchema` here (unless something prevents using it).

Reply:
As far as I can see, we would have to extend `FbTables` to get the relation id

Reply:
Hmm. We can probably leave it as is and see what people need.